app/services/katello/pulp3/repository.rb

```diff
@@ -151,8 +151,9 @@ def delete(href = repository_reference.try(:repository_href))
         end
       end
 
-      def sync
-        [remotes_api.sync(repo.remote_href, repository: repository_reference.repository_href)]
+      def sync(mirror = true)
+        repository_sync_url_data = client_class::RepositorySyncURL.new(repository: repository_reference.repository_href, mirror: mirror)
+        [remotes_api.sync(repo.remote_href, repository_sync_url_data)]
```

Comment:
similarly to pulp2, i'd expect this set in the remote instead of at sync time

Reply:
@jlsherrill : Based on our offline discussion and the fact that we don't have mirror option on remote in pulp3, I'll leave this as is if it's ok to pass this at time of sync.

Reply:
can you just just reference @repo.root.mirror_on_sync here instead of passing it in as a variable?

Reply:
Updated..